### ISSUE TITLE
when handle 16bit grayscale images, DynamicImage::inner_bytes() does not work correctly 

### ISSUE THREAD

**Issue by cloudchamb3r** [image-rs/image#1820]
This happens in the case 

i can't assure that error occures in every 16 bit images, maybe it could be.

using any methods that internally calling DynamicImage::inner_bytes() without matching could cause endianness error

especially in my case, i got this bug when calling `DynamicImage::write_to(&self, w, format)`

## Expected
<img width="407" alt="image" src="https://user-images.githubusercontent.com/98369677/201278931-29ff9ec4-adc2-43e8-a4ae-e566ee676222.png">

## Actual behaviour
<img width="398" alt="image" src="https://user-images.githubusercontent.com/98369677/201279254-84a3dbe1-196d-4b38-ac88-ad7d8f9df2e9.png">

## Reproduction steps

```rust

// not working code
let f = File::create("output.png").unwrap();
let mut w = BufWriter::new(f);
dyn_image.write_to(&mut w, ImageOutputFormat::Png).unwrap();

// working code
let f = File::create("output.png").unwrap();
let mut w = BufWriter::new(f);
match dyn_image {
	| image::DynamicImage::ImageLuma16(dyn_image) => {
		dyn_image
			.write_to(&mut w, ImageOutputFormat::Png)
			.unwrap();
	},

	| _ => todo!(),
}
```

**Comment by fintelia** [comment]
I think this may be a duplicate of https://github.com/image-rs/image/issues/1749 which should be fixed in the 0.24.5 release (published all of 27 minutes ago). Could you give that a try?

**Comment by cloudchamb3r** [comment]
@fintelia  after updating crates, it works. thx ;)
